internal/docker/env.go

```diff
@@ -103,9 +107,9 @@ func ProvideClusterEnv(ctx context.Context, env k8s.Env, runtime container.Runti
 
 	resultEnv, err := overlayOSEnvVars(Env(result))
```

Comment:
nbd but for slightly cleaner code, could have `overlayOSEnvVars` return an `Env` containing the error, if any (rather than returning both an Env and an error and creating the error-env here if err is non-nil)

Reply:
ooh that's a good idea, and makes things simpler